### PR TITLE
Fix url parsing in chat messages

### LINKED ISSUE #1656: Markdown link eats end parenthesis
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Platform - version:**  All - reproducible on v1.0.1-439

**Action Performed (reproducible steps):**
- Log into https://expensify.cash/
- Start chatting with another user
- Send a message in parenthesis with a hyperlink in markdown that has a `/` at end of URL eg `([google](https://google.com/))`

**Expected Result:**
For the URL in parenthesis to hyperlink and for the bracket to close ([google](https://google.com/))

**Actual Result:**
the last parenthetical bracket doesn't appear - 
image

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review @Beamanator 

Fix has been deployed to the `expensify-common` with https://github.com/Expensify/expensify-common/pull/361 & I have updated the dependency here to reflect that.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1656

### Tests / ### QA Steps

   - Log in to https://expensify.cash/
   - Start chatting with another user
   - Send a message in parenthesis with a hyperlink in markdown that has a / at end of URL eg ` ([google](https://google.com/))`

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
 
image

### CODE REVIEW

**Comment by Beamanator** [comment]
@parasharrajat can you fix the merge conflicts?

Maybe related, maybe not - I wonder if there's an issue somewhere since there are 31,846 "changes" to `package-lock.json`

**Comment by parasharrajat** [comment]
Sure

**Comment by parasharrajat** [comment]
@Beamanator Updated. Thanks.

File: package-lock.json [hunk 1]
```diff
@@ -2161,11 +2161,6 @@
             "picomatch": "^2.0.5"
           }
         },
-        "mime": {
-          "version": "2.4.6",
-          "resolved": "https://registry.npmjs.org/mime/-/mime-2.4.6.tgz",
-          "integrity": "sha512-RZKhC3EmpBchfTGBVb8fb+RL2cWyw/32lshnsETttkBAyAUXSGHxbEJWWRXc751DrIxG1q04b8QwMbAwkRPpUA=="
-        },
```
Comment (Beamanator) [inline package-lock.json:2168]: Hmm I see that these lines exist in `master`, is it possible you accidentally removed these but need to put them back? 🤔

Comment (parasharrajat) [reply package-lock.json]: Oops. I did `npm install` and there was an npm version issue so It got removed accidently. I have pushed it back.

**Review by Beamanator — APPROVED** [review]
Looks great! Tested with the new types of links and everything looks great :D

**Pull request merged**